ACadSharp/Color.cs

```diff
@@ -5,297 +5,445 @@
+        public Color(short index)
+        {
+            _color = index;
+        }
+
+        /// <summary>
+        /// Creates a color out of RGB true color bytes.
+        /// </summary>
+        /// <param name="r">Red</param>
+        /// <param name="g">Green</param>
+        /// <param name="b">Blue</param>
+        public Color(byte r, byte g, byte b)
+            : this(new[] { r, g, b })
+        {
+        }
+
+        /// <summary>
+        /// Creates a color out of a RGB true color byte array.
+        /// </summary>
+        /// <param name="rgb">Red Green Blue</param>
+        public Color(byte[] rgb)
+            : this(GetInt24(rgb))
+        {
+
+        }
+
+        /// <summary>
+        /// Creates a color out of a RGB true color.
+        /// </summary>
+        /// <param name="trueColor">True color int 32.</param>
+        private Color(int trueColor)
+        {
+            if (trueColor < 0 || trueColor > MaxTrueColor)
+                throw new ArgumentOutOfRangeException(nameof(trueColor), "True color must be a 24 bit color.");
+                    
+            // Shift to set the 30th bit indicating a true color.
+            _color = trueColor | 1 << 30;
+        }
+
+        /// <summary>
+        /// Creates a color out of a true color int32.
+        /// </summary>
+        /// <param name="color">True color int 32.</param>
+        public static Color FromTrueColor(int color)
+        {
+            return new Color(color);
+        }
+
+        /// <summary>
+        /// Approximates color from a true color RGB.
+        /// </summary>
+        /// <param name="r">Red</param>
+        /// <param name="g">Green</param>
+        /// <param name="b">Blue</param>
+        /// <returns>Approximate AutoCad RGB color.</returns>
+        public static byte ApproxIndex(byte r, byte g, byte b)
+        {
+            var prevDist = -1;
+            for (var i = 0; i < _indexRgb.Length; i++)
+            {
+                var dist = (r - _indexRgb[i][0]) + (g - _indexRgb[i][1]) + (b - _indexRgb[i][2]);
+                if (dist == 0)
+                    return (byte)i;
+
+                if (dist < prevDist)
+                {
+                    prevDist = dist;
+                    return (byte)i;
+                }
+            }
+
+            return 0;
+        }
+
+        /// <summary>
+        /// Returns the RGP color code which matches the passed indexed color.
+        /// </summary>
+        /// <returns>Approximate RGB color from AutoCAD's indexed color.</returns>
+        public static ReadOnlySpan<byte> GetIndexRGB(int index)
+        {
+            return _indexRgb[index].AsSpan();
+        }
+
+        public ReadOnlySpan<byte> GetTrueColorRgb()
+        {
+            if (IsTrueColor)
+            {
+                return new ReadOnlySpan<byte>(BitConverter.GetBytes(_color), 0, 3);
+            }
+
+            return default;
+        }
+
+        public bool Equals(Color other)
```

Comment:
Please add the interface `System.IEquatable<Color>` so the different `System` libraries can use this method.

Reply:
Will do!